### DISCUSSION TITLE
webpack 5 HMR not work

### DISCUSSION THREAD

**Discussion by Idletalk** [webpack/webpack#12944] [Q&A]
here is an example ' https://github.com/Idletalk/webpack5-hmr-test '. at the main feature, HMR is work at webpack 4. but when i upgrade relative packages, such as ' hmr-webpack5 ' feature, HMR is not work. Is there something I missed？

**Comment by knightswatch3** [comment]
Are you using webpack-dev-server ?

**Reply by Idletalk** [reply]
thanks for your reply. i used in devServer at file webpack.config.js. is there something wrong？

**Reply by knightswatch3** [reply]
If achieving the Hot Reload is what you want just pass the necessary args to the webpack-dev server while running with npm. The lesser additional modules the better.

**Reply by Idletalk** [reply]
i have set devServer.hot is true

**Reply by Idletalk** [reply]
the same config，it works at webpack4 but not in webpack5

**Reply by Idletalk** [reply]
the output like this 
![image](https://user-images.githubusercontent.com/38195804/112083087-61c23a80-8bc1-11eb-9fdb-fc231a4b0299.png)

**Reply by alexander-akait** [reply]
You need change the `contentbase` value, right now you `src` and `contenthash` is the same, so webpack reloads page because thinks you change file without HMR

**Reply by Idletalk** [reply]
> You need change the `contentbase` value, right now you `src` and `contenthash` is the same, so webpack reloads page because thinks you change file without HMR

you mean the devServer.contentBase ? but i set the devServer.contentBase webpack-cli shows: 'configuration has an unknown property 'contentBase'. ' the 

- webpack: 5.26.3
- webpack-cli: 4.5.0
- webpack-dev-server: 4.0.0-beta.0

**Reply by alexander-akait** [reply]
for `beta-0` please use the `static` option
